packages/utils/src/internals/robots.ts

```diff
@@ -0,0 +1,126 @@
+import log from '@apify/log';
+// @ts-expect-error This throws a compilation error due to got-scraping being ESM only but we only import types, so its alllll gooooood
+import type { HTTPError as HTTPErrorClass } from 'got-scraping';
+import type { Robot } from 'robots-parser';
+import robotsParser from 'robots-parser';
+import sax from 'sax';
+
+import { gotScraping } from './gotScraping';
+
+let HTTPError: typeof HTTPErrorClass;
+
+export class RobotsFile {
+    constructor(
```

Comment:
Shouldn't the constructor be private too (since it seems like the intended way to use this class is via the static methods)